pkg/kubectl/cmd/apply.go

```diff
@@ -220,7 +230,13 @@ func RunApply(f cmdutil.Factory, cmd *cobra.Command, out io.Writer, options *App
 			helper := resource.NewHelper(info.Client, info.Mapping)
 			patcher := NewPatcher(encoder, decoder, info.Mapping, helper, overwrite)
 
-			patchBytes, err := patcher.patch(info.Object, modified, info.Source, info.Namespace, info.Name)
+			infos := []*resource.Info{info}
+			originalObj, err := resource.AsVersionedObject(infos, false, defaultVersion, encoder)
```

Comment:
Actually, the reason this doesn't have VersionedObject is because you are calling Get() up above.  Why are you calling Get() up above?